compute/cloud-client/src/main/java/compute/preemptible/CreatePreemptibleInstance.java

```diff
@@ -0,0 +1,113 @@
+    // TODO(developer): Replace these variables before running the sample.
+    // projectId: project ID or project number of the Cloud project you want to use.
+    // zone: name of the zone you want to use. For example: “us-west3-b”
+    // instanceName: name of the new virtual machine.
+    String projectId = "your-project-id-or-number";
+    String zone = "zone-name";
+    String instanceName = "instance-name";
+
+    createPremptibleInstance(projectId, zone, instanceName);
+  }
+
+  // Send an instance creation request with preemptible settings to the Compute Engine API
+  // and wait for it to complete.
+  public static void createPremptibleInstance(String projectId, String zone, String instanceName)
+      throws IOException, ExecutionException, InterruptedException {
+
+    String machineType = String.format("zones/%s/machineTypes/e2-small", zone);
+    String sourceImage = String
+        .format("projects/debian-cloud/global/images/family/%s", "debian-11");
+    long diskSizeGb = 10L;
+    String networkName = "default";
+
+    try (InstancesClient instancesClient = InstancesClient.create()) {
+
+      AttachedDisk disk =
+          AttachedDisk.newBuilder()
+              .setBoot(true)
+              .setAutoDelete(true)
+              .setType(AttachedDisk.Type.PERSISTENT.toString())
+              .setInitializeParams(
+                  // Describe the size and source image of the boot disk to attach to the instance.
+                  AttachedDiskInitializeParams.newBuilder()
+                      .setSourceImage(sourceImage)
+                      .setDiskSizeGb(diskSizeGb)
+                      .build())
+              .build();
+
+      // Use the default VPC network.
+      NetworkInterface networkInterface = NetworkInterface.newBuilder()
+          .setName(networkName)
+          .build();
+
+      // Collect information into the Instance object.
+      Instance instanceResource =
+          Instance.newBuilder()
+              .setName(instanceName)
+              .setMachineType(machineType)
+              .addDisks(disk)
+              .addNetworkInterfaces(networkInterface)
+              // Set the preemptible setting.
+              .setScheduling(Scheduling.newBuilder()
+                  .setPreemptible(true)
+                  .build())
+              .build();
+
+      System.out.printf("Creating instance: %s at %s %n", instanceName, zone);
+
+      // Prepare the request to insert an instance.
+      InsertInstanceRequest insertInstanceRequest = InsertInstanceRequest.newBuilder()
+          .setProject(projectId)
+          .setZone(zone)
+          .setInstanceResource(instanceResource)
+          .build();
+
+      // Wait for the create operation to complete.
+      Operation response = instancesClient.insertAsync(insertInstanceRequest).get();
+
+      if (response.hasError()) {
+        System.out.println("Instance creation failed ! ! " + response);
```

Comment:
would it be better to throw an exception here?

Reply:
That would be nice. But we thought of a graceful exception handling and hence followed this style across all previous samples.